### PR TITLE
Drag and Drop: fix bug with dropping snippets in a list

### PR DESCRIPTION
**Pull request by davidrans**
Starting with the second component, form elements can only be dropped
into the second position from the top. Same thing happens when reordering
components within a form.

As far as I can tell from looking at the code this has been broken for a
long time.

QA
----
I adjusted some configuration around which vertical slot a form element
is dropped into based on where the cursor is hovering. Play around with
drag and drop a bit and make sure I didn't break anything else while I
was fixing this bug.

Connects https://github.com/iFixit/dozuki/issues/2157

### CODE REVIEW

**Comment by scbarber** [comment]
CR 💈 Thanks for unpacking that rather dense math! That reads better.

**Comment by andyg0808** [comment]
CR :+1: through fad6fdc

**Comment by davidrans** [comment]
dev_block :hand: on another improvement

**Comment by davidrans** [comment]
un_dev_block :+1:

**Comment by andrewjpiro** [comment]
CR :boom:

File: assets/js/views/my-form.js [hunk 1]
```diff
@@ -41,8 +41,18 @@ define([
 
     , getBottomAbove: function(eventY){
       var myFormBits = $(this.$el.find(".component"));
+      // 30 is an arbitrary pixel offset that experimentally felt right.
+      var cursorY = eventY - 30;
+
+      // If cursor is above the top snippet, there are no snippets to drop it under.
+      if (cursorY < $(myFormBits[0]).offset().top) {
+         return null;
```
Comment (andyg0808) [inline assets/js/views/my-form.js]:
```suggestion
        return null;
```

**Comment by andyg0808** [comment]
CR :+1: through 5efe76f and 7874efa

**Comment by andrewjpiro** [comment]
CR :boom:

**Comment by andyg0808** [comment]
CR :+1: through 97a9f13

**Comment by andrewjpiro** [comment]
CR :boom:

**Comment by k0rvusk0r4x** [comment]
@davidrans Does QA for https://github.com/iFixit/ifixit/pull/32826 count as QA for this?

If not, is set up as simple as clone the repo, check this out, and QA normally, or do I need to do something more?

**Comment by davidrans** [comment]
Nope, I think your QA over there should cover everything here :+1:

**Pull request merged**